### PR TITLE
Support for HTML as output format for report

### PR DESCRIPTION
**Pull request by DahlitzFlorian**
See #14.

It's just the first attempt. Let me know, how you think about it @tonybaloney.

### CODE REVIEW

**Comment by tonybaloney** [comment]
Hi @DahlitzFlorian 

Some early feedback:
- I would’ve utilised either String templates 
https://docs.python.org/3.4/library/string.html#template-strings or a 3rd party templating package. Because you require loops, Jinja2, Mako, or something similar would work. The mixing of html strings and python will rapidly get out of control as more and more people add features to it.
- Reduce the number of font assets in the PR, we should use system fonts where available. I’m not sure on the licensing of those font packages, and also the download for this package should be kept to a minimum
- You’ve accidentally checked in a `.DS_Store` file
- The report function should explicitly check the `—format` value instead of assuming that anything which isn’t console is HTML. There’s plenty of ways to do this, the easiest would be a Python dict with the formats as keys and the callable as the value. aka “the Pythonic switch statement”
- Needs tests :-)
- User should be able to specify the name/path to the output HTML file, using the same flag as the graph command

**Comment by DahlitzFlorian** [comment]
- [ ] Tests
- [ ]  String Templates

**Comment by tonybaloney** [comment]
hi @DahlitzFlorian looks like 2 tests are failing on Windows. please can you fix those first and I'll do a review https://dev.azure.com/AnthonyShaw/wily/_build/results?buildId=46

File: test/unit/test_cli.py [hunk 1]
```diff
@@ -160,6 +161,121 @@ def test_report_with_opts():
...
+                ],
+            )
+            assert result.exit_code == 0, result.stdout
+            assert report.called_once
+            assert check_cache.called_once
+            assert report.call_args[1]["path"] == "foo.py"
+            assert report.call_args[1]["metrics"] == ("example_metric",)
+            assert report.call_args[1]["include_message"]
+            assert report.call_args[1]["n"] == 101
+            assert report.call_args[1]["format"] == "console"
```
[the remark this answers is not in the document]
Comment (DahlitzFlorian) [reply test/unit/test_cli.py]: That was my idea. An appropriate info is logged to stdout in this case. Or do you want to exit with an error message and aborting the creation of a report?

[the remark this answers is not in the document]
Comment (tonybaloney) [reply test/unit/test_cli.py]: I would prefer an explicit error and exit code

**Comment by tonybaloney** [comment]
Let me know when this is ready for another review

**Comment by DahlitzFlorian** [comment]
I’ll let you know. I’m still struggling with the test, but I hope I can fix this by tomorrow.

**Comment by tonybaloney** [comment]
If you need any help can jump on a screen share session and work it out

**Comment by DahlitzFlorian** [comment]
That would be great. Short introduction: Specifying an output path works just fine in every directory and with any project on my machine. However, the corresponding tests fail saying the actual file/directory does not exist even though exit code is 0.
How do you want to share the screen.

**Comment by DahlitzFlorian** [comment]
Just found my mistake: The test failed as it wasn't setup as an integration test. Therefore, the environment was wrong, so no directory was created in the specified one.

**Comment by DahlitzFlorian** [comment]
Sorry for struggling with such small things that long, but I hope that everything is appropriately tested/implemented. Let me know if there is still something not appropriately tested/implemented.

**Comment by tonybaloney** [comment]
This looks great, it's missing a test for `-o` where the name doesn't end in `.html`. But that's no biggie.

Thanks for your contribution! 👍

**Pull request merged**